src/Layers/FeatureLayer/FeatureGrid.js

```diff
@@ -21,6 +21,7 @@ export var FeatureGrid = L.Layer.extend({
 
   onRemove: function () {
     this._map.removeEventListener(this.getEvents(), this);
+    delete this._map;
     this._removeCells();
```

Comment:
that's what triggers [removing](https://github.com/Esri/esri-leaflet/blob/master/src/Layers/FeatureLayer/FeatureLayer.js#L163-L192) the actual geojson features from the featureLayer (and induces errors when the map is still attached)